src/Projection/ProjectorHelper.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Patchlevel\EventSourcing\Projection;
+
+use Patchlevel\EventSourcing\EventBus\Message;
+
+final class ProjectorHelper
+{
+    public function __construct(
+        private readonly ProjectorResolver $projectorResolver = new MetadataProjectorResolver()
+    ) {
+    }
+
+    public function handleMessage(Message $message, Projector ...$projectors): void
+    {
+        foreach ($projectors as $projector) {
+            $handleMethod = $this->projectorResolver->resolveHandleMethod($projector, $message);
+
+            if (!$handleMethod) {
+                continue;
+            }
+
+            $handleMethod($message);
+        }
+    }
+
+    public function createProjection(Projector ...$projectors): void
+    {
+        foreach ($projectors as $projector) {
+            $createMethod = $this->projectorResolver->resolveCreateMethod($projector);
+
+            if (!$createMethod) {
+                continue;
+            }
+
+            $createMethod();
+        }
+    }
+
+    public function dropProjection(Projector ...$projectors): void
+    {
+        foreach ($projectors as $projector) {
+            $dropMethod = $this->projectorResolver->resolveDropMethod($projector);
+
+            if (!$dropMethod) {
+                continue;
+            }
+
+            $dropMethod();
+        }
+    }
+
+    public function handleMessageUsingRepository(Message $message, ProjectorRepository $projectorRepository): void
+    {
+        $this->handleMessage($message, ...$projectorRepository->projectors());
+    }
+
+    public function createProjectionUsingRepository(ProjectorRepository $projectorRepository): void
+    {
+        $this->createProjection(...$projectorRepository->projectors());
+    }
+
+    public function dropProjectionUsingRepository(ProjectorRepository $projectorRepository): void
+    {
+        $this->dropProjection(...$projectorRepository->projectors());
+    }
```

Comment:
Are these really needed? Imho they are not adding much value on top the other methods.

Reply:
Makes it easier to use. And since it's not an interface and just a utility class, I think such small improvements can be added.